indexer/src/configs.rs

```diff
@@ -3,13 +3,17 @@ use std::collections::HashMap;
 
 use crate::errors::{Error, Result};
 
+use serde::Deserialize;
+
 #[derive(clap::Parser, Debug)]
 #[clap(version = "0.0.1")]
 #[clap(subcommand_required = true, arg_required_else_help = true)]
 pub(crate) struct Opts {
     /// Sets a custom config dir. Defaults to ~/.near/
     #[clap(long)]
     pub home_dir: Option<std::path::PathBuf>,
+    #[clap(long)]
+    pub config: Option<std::path::PathBuf>,
```

Comment:
This seems okay, doing it now

Reply:
Other options are either throwing an error if config is specified together with other parameters or making the params override the config. Wdyt?

Reply:
> Other options are either throwing an error if config is specified together with other parameters

My preference here is achieving this with the way `SubCommand` and dependencies structured rather than if-else's.

Reply:
For now kept both as separate, conflicting, `ArgGroups`. I think this is one of the best solutions for now - no separate commands, still safe and no unexpected behavior. We also don't have a use-case for merging configs, so that's okay as well. IMO later changes can be work of another PR if there's the need.